IPython/terminal/pt_inputhooks/qt.py

```diff
@@ -1,10 +1,15 @@
 import sys
 from IPython.external.qt_for_kernel import QtCore, QtGui
 
+# If we create a QApplication, keep a reference to it so that it doesn't get
+# garbage collected.
+_appref = None
+
 def inputhook(context):
+    global _appref
     app = QtCore.QCoreApplication.instance()
     if not app:
-        return
```

Comment:
Want to drop a DeprecationWarning/(UserWarning maybe) here if App does not exists ? so that we can remove it at some point later?